destination/writer/writer_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestWriter_buildUpsertQuery(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		table     string
+		keyColumn string
+		keyValue  any
+		columns   []string
+		values    []any
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want string
+		err  error
+	}{
+		{
+			name: "success",
+			args: args{
+				table:     "users",
+				keyColumn: "id",
+				keyValue:  "someId",
+				columns:   []string{"name", "age", "created", "metadata"},
+				values:    []any{"John", 42, time.Unix(1257894000, 0).UTC(), map[string]string{"action": "insert"}},
+			},
+			want: "MERGE INTO users USING DUAL ON (id = 'someId') " +
```

Comment:
maybe easier to use ` without "

Reply:
Not sure, because in the case with `\``, we have multiline text. But we have to compare it with the single-line result.